lib/model/model.go

```diff
@@ -1567,12 +1568,15 @@ func sendIndexTo(minSequence int64, conn protocol.Connection, folder string, fs
 }
 
 func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo) {
-	m.updateLocals(folder, fs)
-
 	m.fmut.RLock()
 	folderCfg := m.folderCfgs[folder]
+	runner, _ := m.folderRunners[folder]
 	m.fmut.RUnlock()
 
+	// Validate all local changes and update the database
+	fs = runner.validateAndUpdateLocalChanges(fs)
```

Comment:
As commented above, I think this should just be a filter method called before updateLocal gets called from here.